File: examples/calculator-java/server/src/main/java/eth/likespro/lpfcp/calculator/CalculatorImpl.java
package eth.likespro.lpfcp.calculator;

import eth.likespro.lpfcp.LPFCP;

public class CalculatorImpl implements Calculator {
    @Override @LPFCP.ExposedFunction
    public int add(int a, int b) {
        return a + b;
    }

    @Override @LPFCP.ExposedFunction
    public int subtract(int a, int b) {
        return a - b;
    }
}

File: examples/calculator-kotlin/server/src/main/java/eth/likespro/lpfcp/calculator/CalculatorImpl.kt
package eth.likespro.lpfcp.calculator

import eth.likespro.lpfcp.LPFCP

class CalculatorImpl: Calculator {
    @LPFCP.ExposedFunction
    override fun add(a: Int, b: Int): Int = a + b

    @LPFCP.ExposedFunction
    override fun subtract(a: Int, b: Int): Int = a - b
}